cmake/external/CudaRdcUtils.cmake

```diff
@@ -613,24 +642,46 @@ function(cuda_rdc_depends_on OUTVARNAME lib potentialdepend)
       get_target_property(lib_link_libraries ${lib} LINK_LIBRARIES)
       get_target_property(lib_interface_link_libraries ${lib} INTERFACE_LINK_LIBRARIES)
       if(lib_interface_link_libraries)
-        list(APPEND lib_link_libraries lib_interface_link_libraries)
-        list(REMOVE_DUPLICATES lib_link_libraries)
+        if(lib_link_libraries)
+	   list(APPEND lib_link_libraries ${lib_interface_link_libraries})
+        else()
+	   set(lib_link_libraries ${lib_interface_link_libraries})
+	endif()
       endif()
+      list(REMOVE_DUPLICATES lib_link_libraries)
+      string(GENEX_STRIP "${lib_link_libraries}" lib_link_libraries)
     endif()
     if(NOT lib_link_libraries)
+      set_property(TARGET ${lib} APPEND PROPERTY CUDA_RDC_NOT_DEPENDS_ON ${potentialdepend})
       return()
     endif()
-    foreach(linklib ${lib_link_libraries})
+    foreach(linklib IN LISTS lib_link_libraries)
+      cuda_rdc_strip_alias(linklib ${linklib})
       if(linklib STREQUAL potentialdepend)
+        set_property(TARGET ${lib} APPEND PROPERTY CUDA_RDC_DEPENDS_ON ${potentialdepend})
         set(${OUTVARNAME} TRUE PARENT_SCOPE)
         return()
       endif()
+      if(NOT TARGET ${linklib})
+        continue()
+      endif()
+      get_target_property(_finallib ${linklib} CUDA_RDC_FINAL_LIBRARY)
+      # We only consider libraries that might contains RDC Cuda code
+      if(NOT _finallib)
+        continue()
+      endif()
       cuda_rdc_depends_on(${OUTVARNAME} ${linklib} ${potentialdepend})
       if(${OUTVARNAME})
+	if(${${OUTVARNAME}})
```

Comment:
It's not strictly an outvarname then?

Reply:
It is, it set to `TRUE` or `FALSE`

Reply:
For the recursion, I happen to reuse the same name for all levels.

Reply:
But it's really `inout` effectively. Most `outvar` functions set but don't get that variable. It would be good to document too.

Reply:
I still do not see the `inout` nature.  I only see a reuse of a (user given) name in a different scope.

Reply:
On the other hand, i see that
```
      if(${OUTVARNAME})
	if(${${OUTVARNAME}})
```
is actually redundant ... so the code is wrong ...

Reply:
I updated the code to remove the dead/redundant code.

Reply:
That explains it ;) the original code was "if the outvar has a value, and if the variable reference by outvar is true..." i.e. it's using the pointed-to variable.